swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/AbstractResourceTypeHandler.kt

```diff
@@ -516,78 +469,75 @@ abstract class AbstractResourceTypeHandler<T : Resource>(
+      return
+    }
 
-      val confirmedResourcesToDelete = currentMarkedResourcesToDelete.filter { resource ->
-        var shouldSkip = false
-        val candidate = processedCandidates.find { it.resourceId == resource.resourceId }
-        if ((candidate == null || candidate.getViolations().isEmpty()) ||
-          shouldExcludeResource(candidate, workConfiguration, optedOutResourceStates, Action.DELETE)) {
-          shouldSkip = true
-          ensureResourceUnmarked(resource, workConfiguration, "Resource no longer qualifies for deletion")
+    val processedCandidates = candidates
+      .filter { candidate ->
+        currentMarkedResourcesToDelete.any { r ->
+          candidate.resourceId == r.resourceId
         }
-        !shouldSkip
       }
+      .withResolvedOwners(workConfiguration)
+      .also { preProcessCandidates(it, workConfiguration) }
+
+    val confirmedResourcesToDelete = currentMarkedResourcesToDelete.filter { resource ->
+      var shouldSkip = false
+      val candidate = processedCandidates.find { it.resourceId == resource.resourceId }
+      if ((candidate == null || candidate.getViolations().isEmpty()) ||
+        shouldExcludeResource(candidate, workConfiguration, optedOutResourceStates, Action.DELETE)) {
+        shouldSkip = true
+        ensureResourceUnmarked(resource, workConfiguration, "Resource no longer qualifies for deletion")
+      }
+      !shouldSkip
+    }
 
-      val maxItemsToProcess = Math.min(confirmedResourcesToDelete.size, getMaxItemsProcessedPerCycle(workConfiguration))
-      confirmedResourcesToDelete.subList(0, maxItemsToProcess).let {
-        partitionList(it, workConfiguration).forEach { partition ->
-          if (!partition.isEmpty()) {
-            try {
-              deleteResources(partition, workConfiguration)
-              candidateCounter.addAndGet(partition.size)
-            } catch (e: Exception) {
-              log.error("Failed to delete $it. Configuration: {}", workConfiguration.toLog(), e)
-              recordFailureForAction(Action.DELETE, workConfiguration, e)
-            }
+    val maxItemsToProcess = Math.min(confirmedResourcesToDelete.size, getMaxItemsProcessedPerCycle(workConfiguration))
+    confirmedResourcesToDelete.subList(0, maxItemsToProcess).let {
+      partitionList(it, workConfiguration).forEach { partition ->
+        if (!partition.isEmpty()) {
+          try {
+            deleteResources(partition, workConfiguration)
+            candidateCounter.addAndGet(partition.size)
+          } catch (e: Exception) {
+            log.error("Failed to delete $it. Configuration: {}", workConfiguration.toLog(), e)
+            recordFailureForAction(Action.DELETE, workConfiguration, e)
           }
         }
       }
-
-      printResult(
-        candidateCounter,
-        AtomicInteger(currentMarkedResourcesToDelete.size),
-        workConfiguration,
-        markedResources,
-        Action.DELETE
-      )
-    } finally {
-      postClean.invoke()
```

Comment:
why do we not want any of these post actions? anymore

Reply:
They aren't needed. They used to be hooks to let the caller know the task is done to execute something after. This was a side effect of using different threads before. I have been meaning to review these for a bit now.